File: dotnet/Logify.Alert.Core/Collectors/TagsCollector.cs
using System;
using System.Collections.Generic;

namespace DevExpress.Logify.Core.Internal {
    public class TagsCollector : IInfoCollector {
        readonly IDictionary<string, string> tags;

        public TagsCollector(IDictionary<string, string> tags) {
            this.tags = tags;
        }

        public void Process(Exception ex, ILogger logger) {
            if (tags == null || tags.Count <= 0)
                return;

            logger.BeginWriteObject("tags");
            foreach (string key in tags.Keys)
                logger.WriteValue(key, tags[key]);
            logger.EndWriteObject("tags");
        }
    }
}

File: java/Core/src/main/java/com/devexpress/logify/alert/core/collectors/TagsCollector.java
package com.devexpress.logify.alert.core.collectors;

import com.devexpress.logify.alert.core.logger.ILogger;
import java.util.Map;

public class TagsCollector implements IInfoCollector {
    private Map<String, String> tags;

    public  TagsCollector(Map<String, String> tags){
        this.tags = tags;
    }

    public void process(Throwable ex, ILogger logger) {
        if (tags == null || tags.size() == 0)
            return;

        logger.beginWriteObject("tags");
        for (String key : tags.keySet())
            logger.writeValue(key, tags.get(key));
        logger.endWriteObject("tags");
    }
}